db/migrate/20250403234704_backfill_dependencies_new_id.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class BackfillDependenciesNewId < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    # Step 1: Add a unique index; since all values are null, it should be valid already.
+    # This index is needed to make the backfill and the not null constraint efficient.
+    # We will need to manually create this on prod before pushing this migration.
+    add_index :dependencies, :id_new, algorithm: :concurrently, if_not_exists: true, unique: true
+
+    # Step 2: Fill in all null id_new values with a random uuid using a SQL update statement
+    safety_assured do
+      reversible do |dir|
+        dir.up do
+          execute <<-SQL
+            UPDATE dependencies
+            SET id_new = gen_random_uuid()
+            WHERE id_new IS NULL;
+          SQL
```

Comment:
Generating these, updating them, and then the index rebuilding seems like it could exceed the CI timeout. Worth doing in a standalone rake task or something like that, for production?

Reply:
oh it will definitely exceed the timeout. I'm doing it manually before merging this ...